### PR TITLE
Added .NET Core implementation

### PR DESCRIPTION
**Pull request by sabbour**
**This implementation has the following optimizations:**

- Using singleton pattern for both MongoDB and EventHub (AMQP) access
- Using an asynchronous pattern to handle the incoming requests
- Creating a sharded collection and inserting records into 10 partitions
- Potentially using the same implementation for both CosmosDB/MongoDB and RabbitMQ/EventHub

_Note, I moved the Go implementation into its own folder._

### CODE REVIEW

File: netcore/Services/OrderService.cs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                }
+
+                var rnd = new Random(DateTime.Now.Millisecond);
+                int partition = rnd.Next(11);
+                order.Product = $"product-{partition}";
+
+                await ordersCollection.InsertOneAsync(order);
+
+                var db = _isCosmosDb ? "CosmosDB" : "MongoDB";
+                await Task.Run(() =>
```
Comment (sabbour) [inline netcore/Services/OrderService.cs:76]: Will simplify. No need to do this in an async pattern.

File: netcore/Singletons/AMQPClientSingleton.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        // Explicit static constructor to tell C# compiler
+        // not to mark type as beforefieldinit
+        static AMQPClientSingleton()
+        {
+        }
+
+        private AMQPClientSingleton()
+        {
+            // Retrieve the AMQPHost from the Environment Variables
+            _amqpHost = System.Environment.GetEnvironmentVariable("AMQPHOST");
```
Comment (sabbour) [inline netcore/Singletons/AMQPClientSingleton.cs:34]: Will need to update the docs with the environment variable name. I opted to simplify by having a common name AMQPHOST to work with both RabbitMQ and Event Hub.

File: netcore/Singletons/AMQPClientSingleton.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            _eventHubEntity = System.Environment.GetEnvironmentVariable("EVENTHUBNAME");
```
Comment (sabbour) [inline netcore/Singletons/AMQPClientSingleton.cs:35]: Event Hubs need this specifically.

File: netcore/Startup.cs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            var insightsKey = System.Environment.GetEnvironmentVariable("APPINSIGHTS_KEY");
```
Comment (sabbour) [inline netcore/Startup.cs:30]: I found it useful to be able to view my own Application Insights telemetry by providing my own key. Can we send the team tracking telemetry using a different client?

File: netcore/run-docker.sh [hunk 5]
```diff
@@ -0,0 +1,13 @@
+docker build . -t sabbour/captureorderack-netcore:v6
```
Comment (sabbour) [inline netcore/run-docker.sh:1]: This file is not necessary. Could be deleted.

File: netcore/Startup.cs (hunk 4 above)
Comment (shanepeckham) [reply netcore/Startup.cs:30]: We can use two different ENV for a private and public App insights key. We do not expect attendees to have to change code but clearly they can if they want to

**Pull request closed without merging**